src/main/java/uk/gov/hmcts/reform/sscs/service/ccdupdate/CcdLocationUpdateService.java

```diff
@@ -0,0 +1,97 @@
+package uk.gov.hmcts.reform.sscs.service.ccdupdate;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.sscs.ccd.domain.Address;
+import uk.gov.hmcts.reform.sscs.ccd.domain.Hearing;
+import uk.gov.hmcts.reform.sscs.ccd.domain.HearingDetails;
+import uk.gov.hmcts.reform.sscs.ccd.domain.SscsCaseData;
+import uk.gov.hmcts.reform.sscs.ccd.domain.Venue;
+import uk.gov.hmcts.reform.sscs.model.VenueDetails;
+import uk.gov.hmcts.reform.sscs.model.messaging.HmcMessage;
+import uk.gov.hmcts.reform.sscs.service.venue.VenueRpcDetails;
+import uk.gov.hmcts.reform.sscs.service.venue.VenueRpcDetailsService;
+
+import java.util.List;
+import java.util.Optional;
+
+import static java.util.Objects.isNull;
+import static java.util.stream.Collectors.toList;
+
+@Slf4j
+@Service
+public class CcdLocationUpdateService {
+
+    private final VenueRpcDetailsService venueRpcDetailsService;
+
+    public CcdLocationUpdateService(VenueRpcDetailsService venueRpcDetailsService) {
+        this.venueRpcDetailsService = venueRpcDetailsService;
+    }
+
+    public void updateVenue(HmcMessage hmcMessage, SscsCaseData sscsCaseData) {
+
+        Optional<HearingDetails> hearingDetails = getHearingDetailsFromHearingList(hmcMessage, sscsCaseData);
+        if (hearingDetails.isEmpty()) {
```

Comment:
Would this work for the first hearing booked on a case?

Reply:
I believe so, but it will need testing - in such case there will be no venue to remove from CCD , so it will just add the new one with the epimsId from the HMCMessage. Or did you have other scenario in mind?